pkg/lockfile/parse-maven-lock.go

```diff
@@ -116,29 +138,11 @@ func (e MavenLockExtractor) Extract(f DepFile) ([]PackageDetails, error) {
 		return []PackageDetails{}, fmt.Errorf("could not extract from %s: %w", f.Path(), err)
 	}
 
-	details := map[string]PackageDetails{}
-
-	for _, lockPackage := range parsedLockfile.Dependencies {
-		finalName := lockPackage.GroupID + ":" + lockPackage.ArtifactID
-
-		details[finalName] = PackageDetails{
-			Name:      finalName,
-			Version:   lockPackage.ResolveVersion(*parsedLockfile),
-			Ecosystem: MavenEcosystem,
-			CompareAs: MavenEcosystem,
-		}
-	}
+	details := parsedLockfile.collectPackages(parsedLockfile.Dependencies)
 
 	// managed dependencies take precedent over standard dependencies
```

Comment:
I am curious about why managed dependencies take precedent over standard dependencies?

Reply:
Because [they take precedent](https://maven.apache.org/guides/introduction/introduction-to-dependency-mechanism.html#dependency-management):

> * b is defined in B's parent's dependency management section and since **dependency management takes precedence over dependency mediation for transitive dependencies**, version 1.0 will be selected should it be referenced in a or c's POM. b will also have compile scope.
> * Finally, since d is specified in B's dependency management section, should d be a dependency (or transitive dependency) of a or c, version 1.0 will be chosen - again **because dependency management takes precedence over dependency mediation** and also because the current POM's declaration takes precedence over its parent's declaration.

Reply:
my understanding is that dependency management takes precedence
 - for *transitive* dependencies
 - when version requirement is not specified for a direct dependencies

So for the dependencies listed in `with-dependency-management-and-duplicates.xml`, versions of direct dependencies should be reported.

feel free to correct me if this is not accurate!

Reply:
I don't use Maven and don't really do a lot of Java work so I'm only going off the docs and #214 - I agree that it sounds like there could be cases where the behaviour should be different, but I'm not in a good position to accurately confirm.

I think let's open a new issue to discuss and go from there - ultimately, I'm happy to just be told what the logic should be and do the implementation if you've got a better understanding of how Maven is doing things 🙂

(also just to make sure we're on the same page: this is _replacing_ dependencies, not removing them - so the result currently is the _combination_ of both groups of dependencies, and that if the same dependency is in both then the version in `dependencyManagement` wins)

Reply:
sounds good - thank you!